pkg/systemd/services/services.jsx

```diff
@@ -59,22 +59,26 @@ export const systemd_client = {
     user: cockpit.dbus(s_bus.BUS_NAME, { bus: "session" }),
 };
 const timedate_client = cockpit.dbus('org.freedesktop.timedate1');
-export let clock_realtime_now;
-export let clock_monotonic_now;
+export let clock_realtime_now = 0; // ms since epoch; call updateTime() before using
+let monotonic_timer_base = null; // µs
 
 export const MAX_UINT64 = 2 ** 64 - 1;
 
 export function updateTime() {
-    cockpit.spawn(["cat", "/proc/uptime"])
-            .then(function(contents) {
-                // first number is time since boot in seconds with two fractional digits
-                const uptime = parseFloat(contents.split(' ')[0]);
-                clock_monotonic_now = parseInt(uptime * 1000000, 10);
-            }, ex => console.log(ex.toString()));
-    cockpit.spawn(["date", "+%s"])
-            .then(time => {
-                clock_realtime_now = parseInt(time, 10) * 1000;
-            }, ex => console.log(ex.toString()));
+    Promise.all([
+        cockpit.file("/proc/uptime").read(),
+        cockpit.spawn(["date", "+%s"])
+    ])
+            .then(([uptime_contents, date_output]) => {
+                clock_realtime_now = parseInt(date_output, 10) * 1000;
+
+                // first number in /proc/uptime is time since boot (CLOCK_BOOTTIME) in seconds with two fractional digits
+                const uptime = parseFloat(uptime_contents.split(' ')[0]);
+                const clock_boottime_now = parseInt(uptime * 1000000, 10);
+
+                monotonic_timer_base = clock_realtime_now * 1000 - clock_boottime_now;
+            })
+            .catch(ex => console.warn("Failed to read boot time:", ex.toString()));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18466-20230308-073420-5f113059-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)